test/reduce.js

```diff
@@ -35,4 +35,51 @@ describe('reduce', function() {
     eq(sum.length, 1);
   });
 
+  it('Prefers the use of the iterator of an object over reduce (and handles short-circuits)', function() {
+    var symIterator = (typeof Symbol !== 'undefined') ? Symbol.iterator : '@@iterator';
+
+    function Reducible(arr) {
+      this.arr = arr;
+    }
+
+    Reducible.prototype.reduce = function(f, init) {
+      var acc = init;
+      for (var i = 0; i < this.arr.length; i += 1) {
+        acc = f(acc, this.arr[i]);
+      }
+      return acc;
+    };
+
+    Reducible.prototype[symIterator] = function() {
+      var a = this.arr;
+      return {
+        _pos: 0,
+
+        next() {
```

Comment:
Yes, sorry about that... ES6 syntax does get under one's skin after a little while.